src/types/router.ts

```diff
@@ -13,7 +14,20 @@ export type RouterPushOptions = {
   replace?: boolean,
 }
 
-export type RouterPush = (url: string, options?: RouterPushOptions) => Promise<void>
+type RouterPushRoute<
+  TRoutes extends Routes,
+  TRoute extends PropertyKey
+> = RouterPushOptions & {
+  name: TRoute,
+  params?: TRoute extends keyof Flattened<TRoutes> ? Flattened<TRoutes>[TRoute] : Record<never, never>,
+}
```

Comment:
I realized basically typescript didn't like whenever we passed `Flattend<TRoutes>` into another type as the generic. So "flattening" the types fixed the infinite depth error that was in this file.